internal/docstore/example_test.go

```diff
@@ -278,3 +267,59 @@ func ExampleQuery_Update() {
 	// Mel: 20
 	// Pat: 25
 }
+
+func Example_optimisticLocking() {
+	// This example is used in https://gocloud.dev/howto/docstore.
+
+	// Variables set up elsewhere:
+	ctx := context.Background()
+
+	coll, err := memdocstore.OpenCollection("Name", nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer coll.Close()
+
+	// Create a player.
+	pat := &Player{Name: "Pat", Score: 7}
+	if err := coll.Create(ctx, pat); err != nil {
+		log.Fatal(err)
+	}
+	fmt.Println(pat) // memdocstore revisions are deterministic, so we can check the output.
+
+	// Double a player's score. We cannot use Update to multiply, so we use optimistic
+	// locking instead.
+
+	// We may have to retry a few times; put a time limit on that.
```

Comment:
Is this needed even for the memdocstore?

Reply:
Yes, if some other goroutine in the same program is doing the same thing, then they can conflict with each other.